laurent22/joplin
# Desktop: Fixes #9781: Extract default plugins to directories matching their IDs

# Summary

Rather than storing default plugin JPL files in
```
    build/defaultPlugins/[[ plugin ID here ]]/plugin.jpl
```
this pull request stores default plugin JPL files in
```
    build/defaultPlugins/[[ plugin ID here ]].jpl
```

Because `PluginService` uses the name of the `.jpl` file to determine where to extract it, this change fixes #9781.

# Testing

1. Clear Joplin's cache directory (`~/.config/joplin-desktop/cache`).
2. Start Joplin
3. Inspect Joplin's cache directory
    - Verify that there exists a folder named `io.github.jackgruber.backup/` and not a folder named `plugin/`

This has been tested successfully on Ubuntu 23.10.

**Note**: That [the Simple Backup plugin starts successfully is tested by this set of automated tests](https://github.com/laurent22/joplin/blob/dev/packages/app-desktop/integration-tests/simpleBackup.spec.ts).
